src/platform/forms-system/src/js/fields/AutosuggestField.jsx

```diff
@@ -136,11 +136,21 @@ export default class AutosuggestField extends React.Component {
         }
       }
 
-      this.props.onChange(
-        this.props.uiSchema['ui:options'].freeInput || this.useEnum
-          ? inputValue
-          : item,
-      );
+      //
+      const { freeInput, inputTransformers } = this.props.uiSchema[
+        'ui:options'
+      ];
+
+      const inputToSave =
+        inputTransformers &&
+        Array.isArray(inputTransformers) &&
+        inputTransformers.length
+          ? inputTransformers.reduce(
+              (userInput, transformer) => transformer(userInput),
+              inputValue,
+            )
+          : inputValue;
+      this.props.onChange(freeInput || this.useEnum ? inputToSave : item);
```

Comment:
Basically, we're sending a different thing to redux than what we use in local component state (on next line).

In practice, once they hit 'save' they'll see the new string from redux.